src/components/MultiSelectGroup/MultiSelectGroup.stories.tsx

```diff
@@ -0,0 +1,317 @@
+import { VStack } from "@chakra-ui/react";
+import { action } from "@storybook/addon-actions";
+import type { Meta, StoryObj } from "@storybook/react";
+import { useEffect, useState } from "react";
+import { withDesign } from "storybook-addon-designs";
+
+import MultiSelect, {
+  multiSelectWidthsArray,
+} from "../MultiSelect/MultiSelect";
+import MultiSelectGroup from "./MultiSelectGroup";
+import useMultiSelect from "../../hooks/useMultiSelect";
+import { layoutTypesArray } from "../../helpers/types";
+
+const multiSelectItems = [
+  {
+    id: "colors",
+    name: "Colors",
+    items: [
+      { id: "red", name: "Red" },
+      { id: "blue", name: "Blue" },
+      { id: "yellow", name: "Yellow" },
+    ],
+  },
+  {
+    id: "pets",
+    name: "Pets",
+    items: [
+      { id: "cat", name: "Cat" },
+      {
+        id: "dog",
+        name: "Dog",
+        children: [
+          { id: "corgy", name: "Corgy" },
+          { id: "german-sheperd", name: "German Sheperd" },
+          { id: "afghan-hound", name: "Afghan Hound" },
+        ],
+      },
+      { id: "rat", name: "Rat" },
+    ],
+  },
+  {
+    id: "tools",
+    name: "Tools",
+    items: [
+      { id: "hammer", name: "Hammer" },
+      {
+        id: "skrewdriver",
+        name: "Skrewdriver",
+        children: [
+          { id: "slottet", name: "Slotted Skrewdriver" },
+          { id: "phillips", name: "Phillips Skrewdriver" },
+          { id: "allen", name: "Allen Skrewdriver" },
+        ],
+      },
+      { id: "whisk", name: "Whisk" },
+    ],
+  },
+];
+
+const defaultItemsVisible = 5;
+
+const MultiSelectGroupStory = (args) => {
+  const { onChange, onMixedStateChange, onClear, selectedItems } =
+    useMultiSelect();
+
+  // Hack to get storybook's action tab to log state change when selectedItems state changes.
+  const [actionName, setActionName] = useState("");
+
+  useEffect(() => {
+    if (Object.keys(selectedItems).length !== 0) {
+      action(actionName)(selectedItems);
+    }
+    if (actionName === "onClear") {
+      action(actionName)(selectedItems);
+    }
+  }, [actionName, selectedItems]);
+
+  return (
+    <MultiSelectGroup
+      {...args}
+      renderMultiSelect={({ isBlockElement, multiSelectWidth }) => {
```

Comment:
@oliviawongnyc @jackiequach @bigfishdesign13 this is the render prop approach we were chatting about for the two Filter components. Let me know what you think. 

@charmingduchess this approach is also based on the recommendation you gave for the `CheckboxGroup`. The `react.cloneElement` function was removed internally from the `MultiSelectGroup` component and instead this render prop is used to control the the `isBlockElement` and `multiSelectWidth` props. I think something similar can be done for the `CheckboxGroup`.